modules/claims_api/app/controllers/concerns/claims_api/v2/disability_compensation_validation.rb

```diff
@@ -517,6 +540,28 @@ def validate_alternate_names!(service_information)
         end
       end
 
+      def validate_service_branch_names!(service_information)
+        service_information['servicePeriods'].each do |sp|
+          unless brd_service_branch_names.include?(sp['serviceBranch'])
+            raise ::Common::Exceptions::UnprocessableEntity.new(
+              detail: "'servicePeriods.serviceBranch' must match a service branch " \
+                      'returned from the /service-branches endpoint of the Benefits ' \
+                      'Reference Data API.'
+            )
+          end
+        end
+      end
+
+      def brd_service_branch_names
+        @brd_service_branch_names ||= brd_service_branches&.pluck(:description)
+      end
+
+      def brd_service_branches
+        return @brd_service_branches if @brd_service_branches.present?
```

Comment:
we could remove this, that's the point of the instance variable and the `||=`.. it will return `@services_branches`, if not it will make the call and set it to the instance variable, so no reason to check if it's present first.